### ISSUE TITLE
New Fundamentals Overview topic

### ISSUE THREAD

**Issue by tdykstra** [aspnet/Docs#2343]
One thing we don’t have for Fundamentals is an overview topic.  We do have the overview content, but it’s in the Introduction where it’s more detail than is needed for an Intro.  We could reorganize this content by taking sections like “Application Anatomy” out of Introduction and creating a Fundamentals overview article out of them. 

Fundamentals overview would have:
* Application anatomy
* Startup
* Services
* Middleware
* Servers
* Content root
* Web root
* Configuration
* Environments
* Next steps

Introduction would have:
* What is ASP.NET Core?
* Why build ASP.NET Core?
* Build web UI and web APIs using ASP.NET Core MVC
* Client-side development
* Next steps

At the same time we could implement the .NET Core TOC pattern for Fundamentals where the group node goes to index.md rather than just opening and closing the group.

**Comment by GuardRex** [comment]
I guess you feel that a subset of sections would make the cut for the overview, but an alternative is that sections in the overview mirror all of the docs that exist under Fundamentals and appear in the same order as Fundamentals docs.

"Content root" and "Web root:" I guess you're implying that there is enough to say about them to make them separate sections and not part of "Application anatomy." "Application anatomy" would make a great standalone doc, where "content root" and "web root" could be described. Feels like the topic is so *fundamental* that it might appear right at the top of the Fundamentals group of docs to get devs oriented to what's where in an app.

**Comment by tdykstra** [comment]
@GuardRex 

> sections in the overview mirror all of the docs that exist under Fundamentals and appear in the same order as Fundamentals docs.

That would be ideal, but I would go with the current content as a first step in that direction.  Would you be interested in drafting that content?

> "Content root" and "Web root:" 

I was just copying the sections that currently exist without rethinking the organization yet.  You may be right that they don't really fit as sections.

> Feels like the topic is so fundamental that it might appear right at the top of the Fundamentals group of docs to get devs oriented to what's where in an app.

That's exactly what I had in mind with this proposal.

**Comment by tdykstra** [comment]
> and appear in the same order as Fundamentals docs.

As part of this process we should review the present order of the articles as they appear in the Fundamentals TOC node and change the order where it makes sense, rather than just adopting that order for the overview doc.

**Comment by GuardRex** [comment]
> That would be ideal, but I would go with the current content as a first step in that direction. Would you be interested in drafting that content?

Sure, I just need your priority on doing it. I imagine that you want the URL Rewrite doc done first, but let me know. I'm waiting on the outline for URL Rewrite; however, I certainly can guess most of what that doc will need and get most of that doc done soon ... like over the next couple of days.

**Comment by ardalis** [comment]
I'd be interested in working on / collaborating on this, too, when if/when it becomes a priority. I've written most of the Fundamentals articles at this point and would be like to be involved in how they're introduced.

**Comment by GuardRex** [comment]
@ardalis Sounds like its more in your wheelhouse. There's plenty to keep me busy ... URL Rewrite ... others? ... View precomp? You guys can let me know what you need, and I'll get 'er done. :+1:

**Comment by Rick-Anderson** [comment]
@tdykstra  can you split it up first then create an issue to rewrite the overview? 
@ardalis  we'll get you to review the doc.
cc @danroth27
